src/pruner_util.cpp

```diff
@@ -75,20 +75,70 @@ cstring remove_extension(cstring file_path) {
 
 int get_exit_code(cstring name, P4PRUNER::PrunerConfig pruner_conf) {
     INFO("Checking exit code.");
-    cstring command = pruner_conf.validation_bin;
-    // suppress output
-    command += " -i ";
+    if (pruner_conf.err_type == ErrorType::SemanticBug) {
+        cstring command = pruner_conf.validation_bin;
+        // suppress output
+        command += " -i ";
+        command += name;
+        // set the output dir
+        command += " -o ";
+        command += pruner_conf.working_dir;
+        command += " -ll CRITICAL ";
+        command += " -p ";
+        command += pruner_conf.compiler;
+        if (pruner_conf.allow_undef) {
+            command += " -u ";
+        }
+        return WEXITSTATUS(system(command.c_str()));
+    } else {
+        return get_exit_details(name, pruner_conf);
+    }
+}
+
+int get_exit_details(cstring name, P4PRUNER::PrunerConfig pruner_conf) {
+    // The crash bugs variant of get_exit_code
+    cstring command = pruner_conf.compiler;
+    command += " --Wdisable ";
     command += name;
-    // set the output dir
-    command += " -o ";
-    command += pruner_conf.working_dir;
-    command += " -ll CRITICAL ";
-    command += " -p ";
-    command += pruner_conf.compiler;
-    if (pruner_conf.allow_undef) {
-        command += " -u ";
+    // Apparently popen doesn't like stderr hence redirecting stderr to stdout
+    command += " 2>&1";
+
+    char buffer[128];
+    cstring result = "";
+    FILE *pipe = popen(command, "r");
+
+    try {
+        while (fgets(buffer, sizeof buffer, pipe) != NULL) {
+            result += buffer;
+        }
+    } catch (...) {
+        pclose(pipe);
+        throw;
+    }
+    pruner_conf.err_string = result;
```

Comment:
Let's be explicit about when we set the value for this config. So instead return it as a struct. 
```cpp
struct exit_info {
int exit_code;
cstring err_msg
}
```
Something like this.